Add go http server and refactor docker configs/makefile

Jared-Krajewski opened the pull request:
Can probably start creating issues for changes going forward so this doesnt become a mess. 
I will also get an insomnia setup going to save future API calls in. 


This PR does 3 things

1. Adds compose.yaml for api and ollama.
2. Adds a go http server that an query ollama.
3. Because of having compose.yaml and an API to interact with the makefile was cleaned up since it no longer needs a shell script test


To test this run `make build` from project root. 

api can be hit via `POST` to `http://localhost:8080/api/query` with body {"prompt": "test"} 

___ 

This can probably be improved pretty quickly to get the streaming response to work properly, and allow use of additional cores/gpu for response processing. 

right now it can timeout pretty easy (timeout is 60s) and you will see `Error: Timeout was reached` instead of a 200 with a response. If you use insomnia, you will have to go into settings and set the timeout to 60s or it will timeout at 30s by default.

Running this in docker probably isnt the best for an LLM, I'm looking into more options.

Jared-Krajewski committed:
added go http server with api that can query ollama, added compose.yaml to simplify makefile

Jared-Krajewski committed:
increased timeout for http requests to 60s

Jared-Krajewski committed:
updated readme to remove obsolote commans, I will update this in another PR once I have the backend running a bit more optimized since it will probably end up changing again

leekahung reviewed (approved):
This looks good. Was able to curl the model from the terminal after building. The building process itself was also pretty straightfoward. 👍

leekahung commented:
Think we can include `make clean`, `make clobber`, and/or `make restart` commands in the makefile

Jared-Krajewski committed:
added make clean, clobber, restart.. changed model to tinyllama for faster test builds

Jared-Krajewski commented:
Added those in along with `help` in the makefile to explain the commands. I also switched the model to tinyllama to make builds quicker since all of the interactions will otherwise be the same. 

If you want to switch it just sub out `MODEL_NAME=tinyllama` in the makefile, and the env variable in api/compose.yaml 
```
    environment:
      - OLLAMA_MODEL=tinyllama
```
In another pr I will cache the model or allow to pull from local so we don't have to keep re-pulling it. 

I will wait for re-approval or further comments before merging this in.

Jared-Krajewski committed:
fixed end of file newline for makefile

leekahung commented:
> In another pr I will cache the model or allow to pull from local so we don't have to keep re-pulling it. 

Yeah, that would be helpful, thanks!
